### PR TITLE
Finish site (mostly)

### PR DESCRIPTION
**Pull request by alexieyizhe**
Still to be done:

- `mobile` layout
- `smallMobile` layout
- Tweaks to other layout layout
- Form submission logic & backend trigger
- Accessibility
- SEO (site title desc, social preview, etc)

### CODE REVIEW

File: src/sections/Background/index.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import styled from "styled-components";
+
+import BackgroundImageTop from "~static/img/bg_1.svg";
+import BackgroundImageBottom from "~static/img/bg_2.svg";
+import TriangleImg from "~static/img/triangle.svg";
+
+const trianglePositions = [
```
Comment (JamesZhaoLu) [inline src/sections/Background/index.jsx]: For the illustration in the background - did you basically hard-code all of the positions? Were we not able to just use the illustration as a background? Or is this because we want the positions of the triangle to be responsive to the adjustment of the page.

Comment (JamesZhaoLu) [reply src/sections/Background/index.jsx]: Oh wait I see because the triangles must be animated so we need to make them a separate component

File: src/components/Button/index.jsx [hunk 2]
```diff
@@ -13,10 +15,29 @@ const StyledButton = styled.button`
   background-color: ${props =>
     props.theme.colors[props.backgroundColor] || props.backgroundColor};
 
-  // remove annoying blue Chrome outline when focused
   &:focus {
     outline: none;
   }
+
+  & > span {
```
Comment (JamesZhaoLu) [inline src/components/Button/index.jsx]: what type of CSS selector is `& >` ?

File: src/pages/index.jsx [hunk 3]
```diff
@@ -1,39 +1,40 @@
...
-            label="Subscribe"
-            icon="arrow-right"
-            color="white"
-            backgroundColor="primary"
-            type="submit"
-          />
-        </Signup>
-        <Background>heres the background and imgs</Background>
+        <Header />
+        <Main />
```
Comment (JamesZhaoLu) [inline src/pages/index.jsx]: Is there a reason why the Footer component is nested in the Main component? Is this just to make the Foot easily stick to the bottom of the page?

File: src/sections/Background/index.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  height: 60px;
+
+  transform: rotate(${props => props.animDuration * 40 + Math.random() * 50}deg)
+    scale(${props => props.animDuration / 10});
+
+  & > img {
+    width: 100%;
+    animation: float ${props => props.animDuration}s ease-in-out infinite;
+
+    @keyframes float {
```
Comment (JamesZhaoLu) [inline src/sections/Background/index.jsx]: Sorry, I'm not too familiar with the `@keyframes` object? Is this just CSS' way to change based on input (like @media) ?

File: src/sections/Main/MailingListSignup.jsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+  return (
+    <Container>
+      <SignupInput
+        tabIndex="4"
+        type="email"
+        value={email}
+        onChange={handleInputChange}
+      />
+      <SignupButton
```
Comment (JamesZhaoLu) [inline src/sections/Main/MailingListSignup.jsx]: Would the "backend" Code need to be inputted somewhere here once they submit their email for signup?

File: src/components/Button/index.jsx (hunk 2 above)
Comment (alexieyizhe) [reply src/components/Button/index.jsx]: It's an immediate child selector, so this will only target `span` elements that are the immediate children of the current element

For example:
```
<button>
  <span className="willBeTargeted">
    <span className="willNOTBeTargeted">
      some text
    </span>
  </span>
</button>
```
the inner span will not be targeted by this CSS selector

File: src/pages/index.jsx (hunk 3 above)
Comment (alexieyizhe) [reply src/pages/index.jsx]: It's because I was lazy D: We could eventually refactor it out

File: src/sections/Background/index.jsx (hunk 4 above)
Comment (alexieyizhe) [reply src/sections/Background/index.jsx]: `@keyframes` is used to define a CSS animation (see how the name `float` is used in the `animation` style a line up)

It allows you to specify what styles should be applied to the element at a certain time in the animation. In this case, at the start of the animation `0%` the element is in it's normal position (defined by the other base styles on the component). At `45%`, it will have translated down by `10px`, and at `100%` at the end, it'll be back in the normal position. 

This is relative, so whatever duration you set with the `animation` prop, it will adjust and animate according to that duration (and the easing function you set there, which is `ease-in-out` in this case).

File: src/sections/Main/MailingListSignup.jsx (hunk 5 above)
Comment (alexieyizhe) [reply src/sections/Main/MailingListSignup.jsx]: Yes - you can refer to hack the north's implementation for MailingListSignup, but essentially it would be done through an `onClick` prop after you define a function as the callback that will be triggered when they press the button.

For accessibility and user QoL, you should also be looking to make sure the `Enter` key triggers the same callback function when the user is focused or typing in the input field.

File: src/sections/Background/index.jsx (hunk 1 above)
Comment (alexieyizhe) [reply src/sections/Background/index.jsx]: Yep - only the triangles are hard-coded, but right now they're hard coded in the wrong positions since the image will adjust its position dynamically based on the size of the screen. This results in the triangles being in the completely wrong positions on larger screens.

A solution to this would be to change the styling of the triangles relative to the _illustration_, since right now they're relative to the top left corner of the screen right now. This way, we can define a triangle's position to be relative to the centre of the illustration and the illustration doesn't change size, it'll be fine.

tl;dr: image changes position often, but doesn't change size often - position them relative to the illustration and it will look fine

File: gatsby-config.js [hunk 6]
```diff
@@ -6,31 +6,20 @@ module.exports = {
...
     { resolve: `gatsby-plugin-react-helmet` },
+    { resolve: `gatsby-plugin-styled-components` },
     {
       resolve: `gatsby-plugin-alias-imports`,
       options: {
         alias: {
           "~src": "src",
           "~components": "src/components",
           "~pages": "src/pages",
+          "~sections": "src/sections",
```
Comment (alexieyizhe) [inline gatsby-config.js]: this allows us to use absolute imports rather than `import XXX from "../../../../some/stupid/path/../desiredImport.js";`

File: src/components/GlobalStyles/index.jsx [hunk 7]
```diff
@@ -1,8 +1,19 @@
...
 
 const GlobalStyle = createGlobalStyle`
-  html {
-    color: ${props => props.theme.colors.primary};
+  @import url('https://fonts.googleapis.com/css?family=Didact+Gothic&display=swap');
+
+  html, body {
+    font-family: ${props =>
+      props.theme.app
+        .font}, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu,Cantarell, "Helvetica Neue", sans-serif;
```
Comment (alexieyizhe) [inline src/components/GlobalStyles/index.jsx]: Using system fonts allows the site to look good before assets are loaded and, if used as the primary font, makes loading much faster (https://css-tricks.com/snippets/css/system-font-stack/)
